### PR TITLE
Fix pages documentation of theme vars

### LINKED ISSUE #3049: Twig variables 'pages' contains root page, but not the pages object
I recently stumbled across the following issue while tweaking the theme navigation bar and its footer:
```
{{ pages.all|gettype }}
```
Returns NULL

It turns out, that `pages` is not the [Pages Object](https://learn.getgrav.org/16/api#class-gravcommonpagepages) but rather the root of the pages: [`$pages->root()`](https://github.com/getgrav/grav/commit/230a2b594e3f4370db22b5a7dd404a5c9eb665ba#diff-610126884cdfed4d3ac3f346aeb52eb64f1260510b41f41a6adf1d11269fc48dR209).

The [documentation](https://learn.getgrav.org/16/themes/theme-vars#pages-object) however states:

> The pages object represents a nested tree of every page object that Grav knows about.

It also links to the `pages` API, not the `page` API.

Now to solve the issue, we can go two ways:
* **Fix the documentation**
* Fix the code to point to the pages object instead of the root page

The code was added in 2014, so it's been there for quite a long time. The Quark theme for example uses `pages.children.visible` which will fail, if we use the `pages` object, as it required an additional parameter. I guess it would make most sense to fix the documentation? It could make sense to fix this in grav 1.7 as a breaking change.

### PR DESCRIPTION
**Pull request by NicoHood**
Fixes https://github.com/getgrav/grav/issues/3049

### CODE REVIEW

**Comment by mahagr** [comment]
I'll check this when I get back to pages docs.

File: pages/03.themes/06.theme-vars/docs.md [hunk 1]
```diff
@@ -395,9 +395,9 @@ This returns an array of the taxonomy associated with a page.  These can be iter
 
 ### pages object
 
-!! The whole list of the Pages object methods is available on the [API site](https://learn.getgrav.org/api#class-gravcommonpagepages). Here's a list of the methods you'll find most useful.
+!! The Pages object is **[not an instance](https://github.com/getgrav/grav/issues/3049) of the [Pages object](https://learn.getgrav.org/api#class-gravcommonpagepages) but rather the normal [Page Object](https://learn.getgrav.org/api#class-gravcommonpagepage)**. You can apply every function that is listed above in the [Page Object](#page-object). If you want to get a collection of all available pages, you can use `page.collection({'items': '@root.descendants'})` instead.
 
-The **pages object** represents a nested tree of every **page object** that Grav knows about.  This is particularly useful for creating a **sitemap**, **navigation** or if you wish to find a particular **page**.
+The **pages object** represents the **root page object** that Grav knows about. This is particularly useful for creating a **sitemap**, **navigation** or if you wish to find a particular **page**.
```
Comment (mahagr) [inline pages/03.themes/06.theme-vars/docs.md:400]: This is not correct. Pages is just a class which contains various useful methods + logic used to build the page structure.

Comment (mahagr) [reply pages/03.themes/06.theme-vars/docs.md:400]: Oh, I just read your issue. You are right, in twig pages is set to point to the root page, just like you said.

**Comment by mahagr** [comment]
@rhukster I'm not sure what to do with the documentation. In twig, `pages` is really just the root page, which is super confusing.

**Comment by NicoHood** [comment]
What about fixing the doc for 1.6, but fixing the code for 1.7?

**Comment by rhukster** [comment]
> What about fixing the doc for 1.6, but fixing the code for 1.7?

All 1.x releases need to be backwards compatible.  We can’t change the code without breaking that compatibility. We have a responsibility to our users to ensure an upgrade does not break things.

**Comment by NicoHood** [comment]
Okay, but we should at least fix the docs for 1.x then. :-)

**Comment by rhukster** [comment]
Getting to it...

**Comment by mahagr** [comment]
I have fixed the wording.

**Pull request closed without merging**